### PR TITLE
mlearning/tflite-micro: add a config option to redirect micro log to syslog

### PR DESCRIPTION
**Pull request by anchao**
## Summary

mlearning/tflite-micro: add a config option to redirect micro log to syslog

new config option TFLITEMICRO_SYSLOG to redirect micro log to syslog

Signed-off-by: chao an <anchao@lixiang.com>

## Impact

N/A

## Testing

sim/tflm

```
$ cmake -B build -DBOARD_CONFIG=sim/tflm -GNinja
```

```
nsh> tflm_hello
[    3.720000] [ 4] 0 (id=0): size=16, offset=0, first_used=0 last_used=1
[    3.720000] [ 4] 1 (id=1): size=64, offset=64, first_used=1 last_used=2
[    3.720000] [ 4] 2 (id=2): size=64, offset=0, first_used=2 last_used=3
[    3.720000] [ 4] 3 (id=3): size=16, offset=64, first_used=3 last_used=3
[    3.720000] [ 4]  0: 0000000000...................................................................... (1k)
[    3.720000] [ 4]  1: 0000000000..............................1111111111111111111111111111111111111111 (1k)
[    3.720000] [ 4]  2: 22222222222222222222222222222222222222221111111111111111111111111111111111111111 (1k)
[    3.720000] [ 4]  3: 22222222222222222222222222222222222222223333333333.............................. (1k)
[    3.720000] [ 4] 
[    3.720000] [ 4] "Unique Tag","Total ticks across all events with that tag."
[    3.720000] [ 4] FULLY_CONNECTED, 0
[    3.720000] [ 4] "total number of ticks", 0
[    3.720000] [ 4] 
[    3.720000] [ 4] [RecordingMicroAllocator] Arena allocation total 2344 bytes
[    3.720000] [ 4] [RecordingMicroAllocator] Arena allocation head 128 bytes
[    3.720000] [ 4] [RecordingMicroAllocator] Arena allocation tail 2216 bytes
[    3.720000] [ 4] [RecordingMicroAllocator] 'TfLiteEvalTensor data' used 240 bytes with alignment overhead (requested 240 bytes for 10 allocations)
[    3.720000] [ 4] [RecordingMicroAllocator] 'Persistent TfLiteTensor data' used 128 bytes with alignment overhead (requested 128 bytes for 2 tensors)
[    3.720000] [ 4] [RecordingMicroAllocator] 'Persistent buffer data' used 1152 bytes with alignment overhead (requested 1100 bytes for 7 allocations)
[    3.720000] [ 4] [RecordingMicroAllocator] 'NodeAndRegistration struct' used 192 bytes with alignment overhead (requested 192 bytes for 3 NodeAndRegistration structs)
[    3.720000] [ 4] 0 (id=0): size=16, offset=0, first_used=0 last_used=1
[    3.720000] [ 4] 1 (id=1): size=64, offset=64, first_used=1 last_used=2
[    3.720000] [ 4] 2 (id=2): size=64, offset=0, first_used=2 last_used=3
[    3.720000] [ 4] 3 (id=3): size=16, offset=64, first_used=3 last_used=3
[    3.720000] [ 4]  0: 0000000000...................................................................... (1k)
[    3.720000] [ 4]  1: 0000000000..............................1111111111111111111111111111111111111111 (1k)
[    3.720000] [ 4]  2: 22222222222222222222222222222222222222221111111111111111111111111111111111111111 (1k)
[    3.720000] [ 4]  3: 22222222222222222222222222222222222222223333333333.............................. (1k)
[    3.720000] [ 4] 0 (id=0): size=16, offset=16, first_used=0 last_used=1
[    3.720000] [ 4] 1 (id=1): size=16, offset=0, first_used=1 last_used=2
[    3.720000] [ 4] 2 (id=2): size=16, offset=16, first_used=2 last_used=3
[    3.720000] [ 4] 3 (id=3): size=16, offset=0, first_used=3 last_used=3
[    3.720000] [ 4]  0: ................0000000000000000................................................ (1k)
[    3.720000] [ 4]  1: 11111111111111110000000000000000................................................ (1k)
[    3.720000] [ 4]  2: 11111111111111112222222222222222................................................ (1k)
[    3.720000] [ 4]  3: 33333333333333332222222222222222................................................ (1k)
[    3.720000] [ 4] ~~~ALL TESTS PASSED~~~
nsh> poweroff
```

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



No, this PR does not fully meet the NuttX requirements.

* **Summary:** While it describes the change, it lacks crucial information like *why* this change is necessary. Is there a problem with the current logging method? Does this improve performance, debugging, or user experience?
* **Impact:**  "N/A" is insufficient.  While the impact *might* be minimal, you must explicitly state that and justify it.  For instance, does this change the memory usage? Does it add any new dependencies?  Consider each point specifically.  If the answer is truly "NO" for each, then state "NO" and briefly explain why (e.g., "Impact on build: NO, only adds a config option").
* **Testing:** The testing logs don't show a comparison.  You need logs *before* the change using the old logging method, and logs *after* the change demonstrating the redirection to syslog.  Simply providing the steps to build and run the test isn't enough.  Show the actual output difference. Also, specify the details of the `sim/tflm` target.  What version of NuttX are you using? What specific simulator is this (QEMU?)?

The author needs to revise the PR to address these shortcomings before it can be properly reviewed.

File: mlearning/tflite-micro/tflm_syslog.cc [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+#endif
+
+#if !defined(TF_LITE_STRIP_ERROR_STRINGS)
+
+void VMicroPrintf(FAR const char *format, va_list ap)
+{
+  vsyslog(CONFIG_TFLITEMICRO_SYSLOG_LEVEL, format, ap);
+}
+
+void MicroPrintf(FAR const char *format, ...)
```
Comment (xiaoxiang781216) [inline mlearning/tflite-micro/tflm_syslog.cc:37]: why not implement DebugLog like other porting?

**Review by cederom — COMMENTED** [review]
Thank you @anchao :-) Would this also work with Make builds? I can see only CMake updates? :-)

File: mlearning/tflite-micro/tflm_syslog.cc (hunk 1 above)
Comment (xiaoxiang781216) [reply mlearning/tflite-micro/tflm_syslog.cc:37]: tfmico provide the standard method to overwrite the log channel, so it's better to follow their design.

Comment (anchao) [reply mlearning/tflite-micro/tflm_syslog.cc:37]: OK, maybe we can add a NuttX backend

Comment (anchao) [reply mlearning/tflite-micro/tflm_syslog.cc:37]: @xiaoxiang781216 tflm common code adds CRLF processing. If we redirect micro log to the debug log, unnecessary blank lines will appear in the print:

```
[    4.570000] [ 4] 0 (id=0): size=16, offset=0, first_used=0 last_used=1
[    4.570000] [ 4] 
[    4.570000] [ 4] 1 (id=1): size=64, offset=64, first_used=1 last_used=2
[    4.570000] [ 4] 
[    4.570000] [ 4] 2 (id=2): size=64, offset=0, first_used=2 last_used=3
[    4.570000] [ 4] 
[    4.570000] [ 4] 3 (id=3): size=16, offset=64, first_used=3 last_used=3
[    4.570000] [ 4] 
[    4.570000] [ 4]  0: 0000000000...................................................................... (1k)
[    4.570000] [ 4] 
[    4.570000] [ 4]  1: 0000000000..............................1111111111111111111111111111111111111111 (1k)
[    4.570000] [ 4] 
[    4.570000] [ 4]  2: 22222222222222222222222222222222222222221111111111111111111111111111111111111111 (1k)
[    4.570000] [ 4] 
[    4.570000] [ 4]  3: 22222222222222222222222222222222222222223333333333.............................. (1k)
```

Maybe the current implementation is a better choice?

**Comment by anchao** [comment]
> Thank you @anchao :-) Would this also work with Make builds? I can see only CMake updates? :-)


@cederom  thanks for your reminder. I also updated the makefile. Please review again.

File: mlearning/tflite-micro/tflm_syslog.cc (hunk 1 above)
Comment (anchao) [reply mlearning/tflite-micro/tflm_syslog.cc:37]: @xiaoxiang781216

Comment (xiaoxiang781216) [reply mlearning/tflite-micro/tflm_syslog.cc:37]: ok.

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**